Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Gun.h"
+#include "Tile.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    ammunition_ = 1;
+    max_ammunition_ = 1;
+}
+
+bool Gun::UseAmmo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target)
+{
+    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
+    if (tile.valid())
+    {	
+        if(UseAmmo())
+        { 
+            id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+            p->MakeMovementPattern(target,GetOwner());	
+        }
+        else
+        {
+            PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
+        }
```

Comment:
`if(UseAmmo())` -> `if (UseAmmo())` : a space between `if` and `(` is needed).
`id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());`
-> 
`auto projectile = Create<Projectile>(Bullet::T_ITEM_S(), tile);`
You have made `tile`, but you do no use it -> it is needed to use it (because it is shorter and clearer).
Space is needed between function args: `f(a,b)` - wrong, `f(a, b)` - right.
Here it is better to use `auto` instead of `id_ptr_on<Projectile>` because `Create` is template function and returns template type.
Avoid abbreviations: `p` -> 'projectile'.
`PlaySoundIfVisible("empty.ogg",owner.ret_id());` -> `PlaySoundIfVisible("empty.ogg", tile.ret_id());`
Space & owner.ret_id() will not work.